generate_or_validate_json_schemas.py

```diff
@@ -106,6 +107,17 @@ def get_schema_json_dict(cls: Any) -> dict[str, Any]:
         schema_json_dict = TypeAdapter(cls).json_schema(schema_generator=GenerateJsonSchema)
     else:
         raise ValueError(f"Class {cls} is neither a pydantic BaseModel nor an enum.")
+    if {"allOf", "$defs"} == set(schema_json_dict.keys()):
+        assert (
+            len(schema_json_dict["allOf"]) == 1
+        ), "Internal error: Assumed circular reference but structure is unexpected"
+        # This is the case for schemas containing circular references
+        reference_pattern = re.compile(r"^#/\$defs/(?P<cls_name>\w+)$")
+        reference_match = reference_pattern.fullmatch(schema_json_dict["allOf"][0]["$ref"])
+        if reference_match is not None:
+            schema_json_dict_to_merge = schema_json_dict["$defs"][reference_match.group("cls_name")]
```

Comment:
Würde eher nen assert reinhauen, um mypy zu befriedigen. Das macht ein Debugging einfacher, falls es da mal kaputtgehen sollte.
```suggestion
        assert reference_match is not None, "Internal Error: Reference string has unexpected format: {schema_json_dict['allOf'][0]['$ref']}"
        schema_json_dict_to_merge = schema_json_dict["$defs"][reference_match.group("cls_name")]
```
Musst da wahrscheinlich nochmal mit Black drübergehen, wenn du die Suggestion so commitest.